sapi/fpm/fpm/fpm_log.c

```diff
@@ -474,3 +491,36 @@ int fpm_log_write(char *log_format) /* {{{ */
 	return 0;
 }
 /* }}} */
+
+static int fpm_access_log_suppress(struct fpm_scoreboard_proc_s *proc)
+{
+	// Never suppress when query string is passed
+	if (proc->query_string[0] != '\0') {
+		return -1;
+	}
+	// Never suppress if request method is not GET or HEAD
+	if (
+		strcmp(proc->request_method, "GET") != 0
+		&& strcmp(proc->request_method, "HEAD") != 0
+	) {
+		return -2;
+	}
+	// Never suppress when response code is not 200
+	if (SG(sapi_headers).http_response_code != 200) {
+		return -3;
+	}
+	// Never suppress when a body has been sent
+	if (SG(request_info).content_length > 0) {
+		return -4;
+	}
+	struct key_value_s *kv;
+	// Suppress when request URI is an exact match for one of our entries
+	for (kv = fpm_access_suppress_paths; kv; kv = kv->next) {
+		if (kv->value && strcmp(kv->value, proc->request_uri) == 0) {
+			return 1;
+		}
+	}
+	// TODO: Is there something we can set in SG(request_info / sapi_headers) to
+	// flag that the script doesn't want to be logged?
```

Comment:
What would be the point of this? I guess we shouldn't probably allow any request to be hidden if that's what this is about.

Reply:
i agree, wrote in the other PR something about letting files call fpm_suppress_log() but i think it is a Bad Security Idea - will remove this along with the POST tests